browser/ui/brave_shields_data_controller.cc

```diff
@@ -405,6 +424,22 @@ void BraveShieldsDataController::HandleItemBlocked(
     obs.OnResourcesChanged();
 }
 
+void BraveShieldsDataController::HandleItemAllowedOnce(
+    const std::string& allowed_once_type,
+    const std::string& subresource) {
+  GURL subres(subresource);
+  if (allowed_once_type == kJavaScript) {
+    if (resource_list_allowed_once_js_.contains(subres)) {
+      return;
+    }
+    resource_list_allowed_once_js_.insert(std::move(subres));
+  }
+
+  for (Observer& obs : observer_list_) {
+    obs.OnResourcesChanged();
+  }
```

Comment:
I think this loop could be moved into `if (allowed_once_type == kJavaScript) {` block ?

Reply:
done